internal/manifest/auto_optimize.go

```
@@ -0,0 +1,168 @@
+// so an unset one must not read as "unbounded".
+//
+// Read-only, so no `s.mu` (WAL handles concurrent readers).
+func (s *Store) ListAutoOptimizeCandidates(ctx context.Context, limit int) ([]AutoOptimizeCandidate, error) {
+	if limit <= 0 {
+		return nil, nil
+	}
+	rows, err := s.db.QueryContext(ctx, autoOptimizeCandidateSQL, limit)
+	if err != nil {
+		return nil, fmt.Errorf("list auto-optimize candidates: %w", err)
+	}
+	defer rows.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `rows.Close` is not checked

(errcheck)